src/main/java/com/scalyr/integrations/kafka/AddEventsClient.java

```diff
@@ -181,16 +187,20 @@ public AddEventsClient(String scalyrUrl, String apiKey, long addEventsTimeoutMs,
 
   /**
    * Call Scalyr addEvents API with {@link #addEventsTimeoutMs} using exponential backoff.
-   * @param addEventsPayload byte[] addEvents payload
+   * @param addEventsPayload byte[] addEvents payload (post compression if compression is enabled).
+   * @param uncompressedPayloadSize long raw serialized JSON addEvents payload size before the compression.
    * @param startTimeMs time in ms from epoch when `log` is called.  Used to enforce `addEventsTimeoutMs` deadline.
    * @return AddEventsResponse
    */
-  private AddEventsResponse addEventsWithRetry(byte[] addEventsPayload, long startTimeMs) {
+  private AddEventsResponse addEventsWithRetry(byte[] addEventsPayload, long uncompressedPayloadSize, long startTimeMs) {
     log.debug("addEvents payload size {} bytes", addEventsPayload.length);
 
     // 6 MB add events payload exceeded.  Log the issue and skip this message.
-    if (addEventsPayload.length > MAX_ADD_EVENTS_PAYLOAD_BYTES) {
-      log.error("Add events payload size {} exceeds maximum size.  Skipping this add events request.  Log data will be lost", addEventsPayload.length);
+    if (uncompressedPayloadSize > MAX_ADD_EVENTS_PAYLOAD_BYTES || addEventsPayload.length > MAX_ADD_EVENTS_PAYLOAD_BYTES) {
+      // NOTE: compressed size should never really be larger than uncompressed size unless there is a pathological case
+      // and we are trying to compress fully random uncompressable data
+      log.error("Uncompressed add events payload size {} (compressed {}) exceeds maximum size.  Skipping this add events request.  Log data will be lost",
+        uncompressedPayloadSize, addEventsPayload.length);
       if (payloadTooLargeLogRateLimiter.tryAcquire()) {
         log.error("Add events too large payload: {}", new String(addEventsPayload));
```

Comment:
After all these changes for the `uncompressedPayloadSize`, I just realized we are logging the compressed bytes, which won't be very useful.  The use of wrapped output streams make it difficult to get the uncompressed payload and we would pay a performance and memory hit if we did that.  It's probably better just to uncompress the beets before logging them here.  Another alternative would be to just remove this logging and just log/ignore the `line too long` on the server side.